### PR TITLE
Support a value of 0 for eventLimit

### PR DESCRIPTION
**Pull request by acontreras89**
Addresses #2978.

**WIP:**
- [x] Approximate solution
- [x] Get some feedback
- [x] Finish off
- [x] Review specs
- [ ] Update docs?

### CODE REVIEW

**Comment by acontreras89** [comment]
This approximation changes how `eventLimit` works a little. For instance, if we have three events in one day and a limit of 2:

* Before, only one event would be displayed and a +2 badge would be added
* Now, two events are displayed and a +1 badge is added

This was the quickest approach to be able to handle 0 as a valid value for the limit.

I've had to update current tests and the internal default value for `AgendaView`.

File: src/agenda/AgendaView.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import View from '../View'
 import TimeGrid from './TimeGrid'
 import DayGrid from '../basic/DayGrid'
 
-const AGENDA_ALL_DAY_EVENT_LIMIT = 5
+const AGENDA_ALL_DAY_EVENT_LIMIT = 4
```
Comment (arshaw) [inline src/agenda/AgendaView.ts]: it seems that in many places in this PR, the event limit numeric value is now being tested for one-less that it was previously being tested for. did the meaning of "event limit" change or something? trying to understand at a high level

Comment (acontreras89) [reply src/agenda/AgendaView.ts]: If I recall correctly, the old behavior was to display a maximum of `eventLimit` items. For instance, with a limit of 3 the user could see:

* One up to three events, or
* Two events and a *+N* badge

This has changed now, and the limit is inclusive. The user will see three events and the badge.

**Comment by acontreras89** [comment]
There's a bug here that I have already solved in my fork (https://github.com/acontreras89/fullcalendar/commit/5a21d839ffccdea939f891fcb395e727b7da49cc). I'll update the PR when I have the time.

**Comment by acontreras89** [comment]
Updated.

Can anyone give me a hint as to what may be causing CI to fail?

File: src/agenda/AgendaView.ts (hunk 1 above)
Comment (acontreras89) [reply src/agenda/AgendaView.ts]: Sorry I missed this comment.

With an `eventLimit` of `3` and 4 events, the user would see three of them and a badge (+1.) Is that what you were asking?

**Comment by acontreras89** [comment]
@arshaw do we need anything else?

**Comment by arshaw** [comment]
I'm going through old PRs right now.

I will accept this change but only if it is isolated to "always show more-events-popover with eventLimit 0"

As I noted a while back, this also changed the definition of `eventLimit` to be one-less. I believe these other issues match this request as well: #3129, #3035. I need to think about that API change more.

**Comment by acontreras89** [comment]
> I will accept this change but only if it is isolated to "always show more-events-popover with eventLimit 0"

There's no easy way to isolate this. Code has things like `td = cellMatrix[levelLimit - 1][seg.leftCol]`, which will crash if we feed a value of 0.

This is because `eventLimit` is conceptually (and semantically) wrong. For a value of _N_ and more than _N_ events, the _Nth_ event is being hidden. But hidding the 0th event doesn't make any sense, does it?

**Comment by arshaw** [comment]
closing per my comment here:https://github.com/fullcalendar/fullcalendar/issues/2978#issuecomment-647795671

thanks @acontreras89

**Pull request closed without merging**